proposal.rst

```diff
@@ -267,29 +285,29 @@ Project Timeline
 .. any university classes or exams, vacations, etc), make sure you include them
 .. in your project timeline.
 
-* Weeks of bonding: get familiar with ranking algorithms, search for suitable datasets, assessment of the complexity of integrating large databases, and performance tests.
 
-In each next position assumes that testing and writing cross-platform part is included.
+* Weeks of bonding: Get more familiar with ranking algorithms, select specific suitable parts of the above datasets, discuss what is wrong with `#278 <https://github.com/xapian/xapian/pull/278>`_.
 
-* 1 Week - integrate datasets 
+* 1 Week - FIRE dataset integration. Each user have to download it by themself, because of 
+registration required. Implement training process using Letor API.  
 
-* 2 Weeks of exams (situation with exams is very unstable now in my University, so I timeline may change in this case)
+* 2 Weeks of exams (situation with exams is very unstable now in my University, so I timeline may change in this case).
 
-* 1 Week - solve possible problems with the size of datasets
+* 1 Week - add function for selecting scorer and ranker. It would be nice to write API allows user to select default sets of queries to run and run specified queries. For that purpose update build system to be able to run suite from terminal.
 
-* 3 Weeks - implement ListMLE, ListNet and SVM evaluation api
+* 1 Week - TREC dataset integration. It has similar format to FIRE, it should not be problem to integrate it. Start writing tool for combine docs.
 
-If no troubles appeared, I can try to develop suite for performance testing:
+* 1 Week - combine tool API should support plain logical operations, it could take time to implement them.
 
-* 1 Week - get familiar with data indexing and searching api
+* 1 Week - prepare tool for text data extracting from Wikipedia.
 
-* 1.5 Weeks - tests for indexing
+* 1 Week - add functionallity to perftest to drop system cache for more accurate results and run one case multiple times to get average time consuming.
 
-* 1.5 Weeks - tests for searching
+* 1 Week - implement Wikipedia data loader and queries for it, write performance tests for filling db with documents, iterating over documents.
 
-* 1 Week - solving problems and integrate tests to build system
+* 1 Week - add perftest to autotools srcipt, as it advised in `#107 <https://trac.xapian.org/ticket/107>`_. If time is left, add function to fork test case in new process and measure it memory usage.
```

Comment:
Why do you need to fork to measure memory usage? Can you not use `getrusage()` or similar within the performance test? Or are you worried about significant overhead from the performance test itself? Are you going to check memory usage several times and graph it? How are you going to report what you find?